src/data/automation_i18n.ts

```diff
@@ -653,6 +653,67 @@ export const describeCondition = (
     return base;
   }
 
+  // Time condition
+  if (condition.condition === "time") {
+    const weekdaysArray = ensureArray(condition.weekday);
+    const validWeekdays =
+      weekdaysArray && weekdaysArray.length > 0 && weekdaysArray.length < 7;
+    if (condition.before || condition.after || validWeekdays) {
+      const before = condition.before?.toString().includes(".")
+        ? `entity ${
+            hass.states[condition.before]
+              ? computeStateName(hass.states[condition.before])
+              : condition.before
+          }`
+        : condition.before;
```

Comment:
Bonus : It can be nice to format the hour in the locale format (with am pm or not)

Reply:
If you want to merge this I can come back later and add localization to both time trigger and time condition. Currently they are consistent in how they display the time. May or may not get to it before the next release cycle.